benchmarks/asv_delegated.py

```diff
@@ -0,0 +1,350 @@
+        """Create the symlink to the delegated environment."""
+        self._path_delegated.unlink(missing_ok=True)
+        self._path_delegated.parent.mkdir(parents=True, exist_ok=True)
+        self._path_delegated.symlink_to(delegated_env_path, target_is_directory=True)
+        assert self._delegated_found
+
+    def _setup(self):
+        """Temporarily try to set the user's active env as the delegated env.
+
+        Environment prep will be run anyway once ASV starts checking out
+        commits, but this step tries to provide a usable environment (with
+        python, etc.) at the moment that ASV expects it.
+
+        """
+        current_env = Path(sys.executable).parents[1]
+        message = (
+            "Temporarily using user's active environment as benchmarking "
+            f"environment: {current_env} . "
+        )
+        try:
+            self._symlink_to_delegated(current_env)
+            _ = self.find_executable("python")
+        except Exception:
+            message = (
+                f"Delegated environment {self.name} not yet set up (unable to "
+                "determine current environment)."
+            )
+            self._path_delegated.unlink(missing_ok=True)
+
+        message += "Correct environment will be set up at the first commit checkout."
+        log.warning(message)
+
+    def _prep_env(self, repo: Repo, commit_hash: str) -> None:
+        """Prepare the delegated environment for the given commit hash."""
+        message = (
+            f"Running delegated environment management for: {self.name} "
+            f"at commit: {commit_hash[:8]}"
+        )
+        log.info(message)
+
+        env_prep: EnvPrepCommands
+        try:
+            env_prep = self._env_prep_lookup(repo, commit_hash)
+        except KeyError as err:
+            message = f"Problem finding env prep commands: {err}"
+            log.error(message)
+            raise EnvironmentUnavailable(message)
+
+        new_env_per_commit = self.COMMIT_ENVS_VAR in environ
+        if new_env_per_commit:
+            env_parent = env_prep.env_parent / commit_hash[:8]
+        else:
+            env_parent = env_prep.env_parent
+
+        # See :meth:`Environment._interpolate_commands`.
+        #  All ASV-namespaced env vars are available in the below format when
+        #  interpolating commands:
+        #   ASV_FOO_BAR = {foo_bar}
+        # We want the env parent path to be one of those available.
+        global_key = f"ASV_{EnvPrepCommands.ENV_PARENT_VAR}"
+        self._global_env_vars[global_key] = str(env_parent)
+
+        # The project checkout.
+        build_dir = Path(self._build_root) / self._repo_subdir
+
+        # Run the script(s) for delegated environment creation/updating.
+        # (An adaptation of :meth:`Environment._interpolate_commands`).
+        for command, env, return_codes, cwd in self._interpolate_commands(
+            env_prep.commands
+        ):
```

Comment:
Following [the pattern in ASV's own source](https://github.com/airspeed-velocity/asv/blob/8877eff343807e8bc594690dec08bc0da2caa401/asv/environment.py#L821).